### ISSUE TITLE
Expand EICExtractor

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#7088]
For quality control purposes, it would be great to get additional information on the extracted peak(s) as provided by FeatureFinderMetaboIdent.

E.g.,: monitoring a peptide:

```
CompoundName	SumFormula	Mass	Charge	RetentionTime	RetentionTimeRange	IsoDistribution
LVNELTEFAK	C53H86N12O17	0	2	810	0	0
```
can be achieved with
```
FeatureFinderMetaboIdent -in 190215_Q_QC01_01_32.mzML -id 190215_Q_QC01_01_32_wo_formula.tsv -out 190215_Q_QC01_01_32.featureXML -model:type none
TextExporter -in ./190215_Q_QC01_01_32.featureXML -out test.tsv -feature:add_metavalues 100
```

This functionality is highly useful but equally well hidden from the user ;)

User feedback suggested putting this functionality in the EICExtractor tool

**Comment by timosachsenberg** [comment]
Hi @rolivella I added a quick PR that adds another column with the area here:
https://github.com/OpenMS/OpenMS/pull/7115
Do you have some small test input that I could add as a test?

**Comment by rolivella** [comment]
Hi @timosachsenberg , thanks! Here a link were you can download the file: 

https://www.dropbox.com/scl/fi/e6fp8jnsmaomv5bzd75pb/190215_Q_QC01_01_32.mzML?rlkey=id8kb8gh5t5qsrtjpdhbu1mp6&dl=0

I've tried to filter it by mass by the result was not as expected and for this I leave you with the whole file. 

If you need something else just tell me.

**Comment by timosachsenberg** [comment]
The other input and parameters you use to run EICExtractor would be great

**Comment by rolivella** [comment]
@timosachsenberg yes, first a print screen of XCalibur showing the peak that should be detected (the one that has a peak area of 2.56E9) and that the FeatureFinderMetaboIdent is able to identify: 

![221206456-7ac1e14f-c9b0-439f-8156-f579fb65a677](https://github.com/OpenMS/OpenMS/assets/1679820/0cc3a870-76d0-4083-a64c-3e4e2e226737)

These are the parameters: 
```tsv
CompoundName	SumFormula	Mass	Charge	RetentionTime	RetentionTimeRange	IsoDistribution
LVNELTEFAK		1162.62338	2	810	0	0
```

**Comment by rolivella** [comment]
Hi @timosachsenberg ! Did you manage to test the PR?

**Comment by timosachsenberg** [comment]
I am totally swamped with other tasks and need to be practical here. So I will just push it out in our container only release 3.1 (hopefully this week) and leave the testing to you or someone else. If it doesn't work, I will fix it and add tests later. I hope that works for you

**Comment by rolivella** [comment]
@timosachsenberg were you able to push the release 3.1? Thanks!

**Comment by timosachsenberg** [comment]
for conda it is already published.
for our hosted containers they should be ready once those ran through: https://github.com/OpenMS/OpenMS/actions/runs/6612269061

**Comment by rolivella** [comment]
@timosachsenberg It does not work, I don't know if I'm doing something wrong: 

```bash
proteomics@hipnos6:~/mysoftware/openms/EICExtractor-openms310$ singularity exec /home/proteomics/mysoftware/openms/build/ghcr.io-openms-openms-executables-3.1.0.img EICExtractor -in 190215_Q_QC01_01_32.mzML -pos 190215_Q_QC01_01_32.edta -out 190215_Q_QC01_01_32.csv
Error: Unable to read file (Failed parsing in line 2: Could not convert the first three columns to a number!
Offending line: '2022.0  582.3189    0'  (line 2)
 in: )
Detected a header line.

```
Where 

```bash
proteomics@hipnos6:~/mysoftware/openms/EICExtractor-openms310$ cat 190215_Q_QC01_01_32.edta
RT  m/z int
2022.0  582.3189    0

```

**Comment by timosachsenberg** [comment]
thanks for the feedback. will check

**Comment by timosachsenberg** [comment]
File format seems to be a bit picks. If I only use spaces, I get it to run with:
```
RT m/z Intensity
2022.0 582.3189 0.0
```
But: RT of 2022.0 s is far from the actual peak (the file I have is not even that long).

If I extract a huge RT range with: EICExtractor -in e.mzML -pos e.edta -out test.csv -rt_tol 5000.0 -mz_tol 20.0
I get:
```
/ceph/ibmi/abi/projects/sachsenb/OpenMS/openms-build/roger$ cat test.csv
,,e.mzML,,,,,
,,,,,,,
RT,mz,RTobs,dRT,mzobs,dppm,intensity,area
2022.0,582.318899999999985,762.79396494401999,-1259.20603505598001,582.319266684965783,-0.998748166979093,6.517387e08,1.001493667492654e10
```
Which seems to look good (has the area column)

**Comment by rolivella** [comment]
Yes, sorry, with the command: 

`singularity exec /home/proteomics/mysoftware/openms/build/ghcr.io-openms-openms-executables-3.1.0.img EICExtractor -in 190215_Q_QC01_01_32.mzML -pos 190215_Q_QC01_01_32.edta -out 190215_Q_QC01_01_32.csv`

and 

```bash
proteomics@hipnos6:~/mysoftware/openms/EICExtractor-openms310$ cat 190215_Q_QC01_01_32.edta
RT m/z int
762 582.3189 0
```

I get: 

```bash
proteomics@hipnos6:~/mysoftware/openms/EICExtractor-openms310$ cat 190215_Q_QC01_01_32.csv
,,190215_Q_QC01_01_32.mzML,,,,,
,,,,,,,
RT,mz,RTobs,dRT,mzobs,dppm,intensity,area
762.0,582.318899999999985,762.79396494401999,0.79396494401999,582.31884765625,-0.089888461434249,6.612977e08,2.01544625e09
```

So 2.01544625e09 which is almost the area we see at XCalibur: https://github.com/OpenMS/OpenMS/issues/7088#issuecomment-1752985174 which is 2.56E9. 

For us this result is OK so you can close the issue. 

Many thanks!

**Comment by rolivella** [comment]
BTW, if I put 

`-rt_tol 240 -mz_tol 5`

I get 2.52e09 which is even closer to 2.56E9 from XCalibur!

**Comment by timosachsenberg** [comment]
Great and thanks for the feedback! Do you know how large the RT window extracted by XCalibur is? My guess is that this is the major source of differences.

**Comment by timosachsenberg** [comment]
Closing the issue but feel free to continue discussion

**Comment by rolivella** [comment]
Hi @timosachsenberg , I'm testing the EICExtractor with another QC file and I cannot obtain exactly the same numbers. The RAW and mzML files are here: https://www.dropbox.com/scl/fo/rf1tdjqylei3lnn7n3tso/h?rlkey=vqnhh5j8gsz26p6iqg353euxi&dl=0

Here the area extracted by XCalibur: 

![Screenshot from 2023-10-30 12-07-21](https://github.com/OpenMS/OpenMS/assets/1679820/4da5b3d1-9684-4afb-be9d-51454d8ed01d)

The RT is 14.4min (864sec) and the area is 1.23E09. 

I did the XIC with EICExtractor with this EDTA: 

```
RT m/z int
864 582.3189 0
```

And obtained 2.293E09 with RT_tol of 240 sec and RT_mass 5 ppm. 

I did a third calculation with this tool https://github.com/fgcz/rawrr (with RT_tol of 5 ppm) and I got 1.24E09. 

So in summary: 

Area extracted with: 
* XCalibur:  1.23E09
* EICExtractor: 2.293E09
* rawrr: 1.24E09 

Do you know why this difference between XCalibur and EICExtractor?

**Comment by timosachsenberg** [comment]
Just one guess: is RT tolerance and or m/z tolerance calculated differently (e.g., +- tolerance or +- 0.5 * tolerance)

If that is not the case: @hroest or @jcharkow I think I would need some input here.
I basically just call the OpenSwath methods.

**Comment by rolivella** [comment]
I'm not sure but I tested with rawrr: 
* mass tol 2.5 ppm, and area=8.58E8 
* mass tol 10 ppm, and area=1.25E9

With XCalibur we tested different mass tol and give us the same.

**Comment by timosachsenberg** [comment]
can you e.g. also half the tolerances in EICExtractor (also rt). Because it could be that EICExtractor calculated the extraction window differently

**Comment by timosachsenberg** [comment]
e.g. it could be that EICExtractor extracts double the RT range which could explain the higher signal

**Comment by rolivella** [comment]
With EICExtractor, I obtain: 
* MZ_tol 2.5 ppm, RT_tol 120 sec, area = 2.285516077179687e09
* MZ_tol 10 ppm, RT_tol 120 sec, area = 2.293877119226563e09

**Comment by jcharkow** [comment]
Sorry I'm not too sure, maybe @hroest  has a better idea?

**Comment by rolivella** [comment]
Could it be that EICExtractor is integrating MS1 and MS2 and for this reason the area is higher than expected?

**Comment by timosachsenberg** [comment]
hmm was not the case in the past but I think the logic behind areaiterator changed recently (will consider the first matching spectrum level).
I will provide a quick fix that only loads MS1 in the first place. https://github.com/OpenMS/OpenMS/pull/7169

**Comment by rolivella** [comment]
Ideally the user could run the tool for MS1 or MS2, this would avoid to use first FileFilter

**Comment by timosachsenberg** [comment]
ok need to think about that... but did you get better results if you just keep the MS1 (with FileFilter) and then try to run EICExtractor?

**Comment by rolivella** [comment]
Filtering the mzML selecting only MS1 I get the same area=2.293877119226563e09

Maybe it's not this then...

**Comment by jpfeuffer** [comment]
Have you tried playing around with RT_tol?

**Comment by rolivella** [comment]
> Have you tried playing around with RT_tol?

https://github.com/OpenMS/OpenMS/issues/7088#issuecomment-1785485809

**Comment by timosachsenberg** [comment]
my R is a bit rusty but looking at the rawrr code it looks like it might stop chromatogram extraction once it hits zero or intensities increase again... I need to look at the data. We also have such an option, but it is currently not enabled.

**Comment by jpfeuffer** [comment]
> > Have you tried playing around with RT_tol?
> 
> [#7088 (comment)](https://github.com/OpenMS/OpenMS/issues/7088#issuecomment-1785485809)

Ah sorry, since both RT_tols in the comment were the same, I thought you might not have tried, yet. But comparing to your initial settings you halved the tolerance as suggested.

**Comment by timosachsenberg** [comment]
k I think I might have found the issue:
I zoomed in on this peak and there are a lot of masses in the tolerance area.
I currently also sum up all masses in the tolerance window per spectrum but it seems a better / more established way is to just take the maximum peak in the tolerance window.
I can provide a fix
![image](https://github.com/OpenMS/OpenMS/assets/5803621/4fb758e8-e981-47e9-b2d1-208d06155d3c)

**Comment by rolivella** [comment]
Good! Have you tried to just take the maximum peak and see if the area is nearer XCalibur/rawrr?

**Comment by timosachsenberg** [comment]
yeah just tried but doesn't seem to be the reason. Will check if we need to stop extraction if we hit the baseline

**Comment by timosachsenberg** [comment]
It currently extracts these intensities. if run with -rt_tol 120 -mz_tol 2.0. but not sure if removing this small bump is enough...
```
816.046 248299
848.549 139188
849.117 197028
850.725 388046
851.22  1.02234e+06
851.701 1.3656e+06
852.193 2.39602e+06
852.663 1.6047e+06
853.176 1.96251e+06
853.672 1.21442e+06
854.181 1.74225e+06
854.71  1.68234e+06
855.836 809336
856.401 764181
856.976 766641
857.544 467501
858.112 398804
858.653 644380
859.171 261084
860.205 375625
861.273 190495
861.81  1.18159e+06
862.361 1.64673e+07
862.923 7.52557e+07
863.508 1.74238e+08
864.063 2.46144e+08
864.574 2.34862e+08
865.097 2.43821e+08
865.603 2.2679e+08
866.123 1.91433e+08
866.667 1.6602e+08
867.228 1.68316e+08
867.796 1.09345e+08
868.388 8.87694e+07
868.98  6.28508e+07
869.592 4.51123e+07
870.18  3.33019e+07
870.796 2.39865e+07
871.392 1.95475e+07
871.999 1.67111e+07
872.611 1.27466e+07
873.22  9.56981e+06
873.824 7.8379e+06
874.427 7.4955e+06
874.927 5.86935e+06
875.818 6.31003e+06
876.242 1.0716e+07
878.012 3.25033e+06
878.487 4.76632e+06
879.908 2.5418e+06
880.403 2.46655e+06
880.932 1.75954e+06
881.452 1.74996e+06
882.016 1.67494e+06
882.578 1.19016e+06
883.168 1.33875e+06
883.76  1.38718e+06
884.376 1.62231e+06
889.058 1.57639e+06
889.486 906986
894.244 432041
901.718 221375
903.127 161779
904.673 95489
905.31  104568
907.781 162917
911.353 137392
912.525 170941
```

**Comment by timosachsenberg** [comment]
I think the issue are indeed the boundaries. The RT range is pretty large, and it will integrate over the small peaks as well.
I think someone with better OpenSWATH knowledge should jump in here and suggest a fix. I don't have capacities right now to dig into it.

**Comment by jcharkow** [comment]
I believe the OpenSwath peak integrator is just summing all of the points across the specified retention time range. I do not think any peak picking is occurring. Not sure if peak picking should be occurring in this application?

**Comment by timosachsenberg** [comment]
@jcharkow thanks! This makes sense. I think we could make it an option in EICExtractor to also pick the peak. Would be good to see how close we are to Excalibur or rawer on orbitrap with our existing algorithms. Which openswath algorithm and settings would be the something close to what they are doing?

**Comment by jcharkow** [comment]
I think we would probably want to use the `PeakPickerMRM` class specifically the `pickChromatogram()` function which provides the left and right boundaries as well as the apex intensity and area intensity for all peaks found in the area. Usually with OpenSwath I use the default settings for OpenSwathWorkflow (Note: these are different from the class's default settings) which uses svgolay filter frame_length 11 and polynomial order of 3. However these may have to be fine-tuned as they are usually performed at the MS2 level.

**Comment by jpfeuffer** [comment]
I think so, too. It also has PeakIntegrator settings to fit models.
We use it in the DDA featurefinders, too.

**Comment by jpfeuffer** [comment]
Oh I actually confused it with FFMRM which uses MRMTransitionGroupPicker which then uses a combination of PeakPickerMRM (which only has basic integration) and PeakIntegrator (which does elaborate integration with background subtraction)

**Comment by timosachsenberg** [comment]
Does anyone know by chance if Excalibur does peakpicking?
What I try to understand is if it is reasonable to implement the peak picking in EICExtractor e.g. in addition to just calculating the total area without considering peak boundaries.
It sounds to me that the picked peak area is more useful for QC? @rolivella 
And if peak picking makes sense - can we assume this works for a wider range of vendors if we implement it @jcharkow .

**Comment by timosachsenberg** [comment]
@jcharkow I tried your defaults with PeakPickerMRM.
What do you think would be good boundaries here?
Also, abundance from PeakPickerMRM (as stored in FloatDataArray) is still much higher.

```
Picked intensity:2.40947e+08    abundance: 2.21271e+09      left: 861.81    right: 878.487
816.046 248299
848.549 139188
849.117 197028
850.725 388046
851.22  1.02234e+06
851.701 1.3656e+06
852.193 2.39602e+06
852.663 1.6047e+06
853.176 1.96251e+06
853.672 1.21442e+06
854.181 1.74225e+06
854.71  1.68234e+06
855.836 809336
856.401 764181
856.976 766641
857.544 467501
858.112 398804
858.653 644380
859.171 261084
860.205 375625
861.273 190495
861.81  1.18159e+06
862.361 1.64673e+07
862.923 7.52557e+07
863.508 1.74238e+08
864.063 2.46144e+08
864.574 2.34862e+08
865.097 2.43821e+08
865.603 2.2679e+08
866.123 1.91433e+08
866.667 1.6602e+08
867.228 1.68316e+08
867.796 1.09345e+08
868.388 8.87694e+07
868.98  6.28508e+07
869.592 4.51123e+07
870.18  3.33019e+07
870.796 2.39865e+07
871.392 1.95475e+07
871.999 1.67111e+07
872.611 1.27466e+07
873.22  9.56981e+06
873.824 7.8379e+06
874.427 7.4955e+06
874.927 5.86935e+06
875.818 6.31003e+06
876.242 1.0716e+07
878.012 3.25033e+06
878.487 4.76632e+06
879.908 2.5418e+06
880.403 2.46655e+06
880.932 1.75954e+06
881.452 1.74996e+06
882.016 1.67494e+06
882.578 1.19016e+06
883.168 1.33875e+06
883.76  1.38718e+06
884.376 1.62231e+06
889.058 1.57639e+06
889.486 906986
894.244 432041
901.718 221375
903.127 161779
904.673 95489
905.31  104568
907.781 162917
911.353 137392
912.525 170941
```

**Comment by timosachsenberg** [comment]
Another experiments:
- used left: 861.81    right: 878.012 as determined by PeakPickerMRM in PeakIntegrator
result: 2.224882589e09

still too high...

**Comment by timosachsenberg** [comment]
k I think I found the right combination now.
I just went back and integrate the EIC with trapezoid and get 1.264e09.
My mistake was that I thought this is the default, but intensity summing was set as default.
So no boundary detection is performed right now...

**Comment by timosachsenberg** [comment]
https://github.com/OpenMS/OpenMS/pull/7177

**Comment by timosachsenberg** [comment]
To summarize:
We use standard trapezoid integration over the full extraction window. This seems to be very close to what thermo or rawrr is doing. Let me know if that works for you. Containers should be online now.

**Comment by rolivella** [comment]
Ok, thank you very much,  I'll test it and let you know ASAP.

**Comment by rolivella** [comment]
Hello @timosachsenberg, I made some test with the last QC file and I'm a bit confused, maybe I'm not using the right container version. 

From this https://github.com/OpenMS/OpenMS/issues/7088#issuecomment-1785172607 with XCalibur I obtained an area of 1.2E9 aprox, but using this two last versions I obtained this numbers: 

```
Version: 3.1.0-pre-nightly-2023-11-02 Nov  3 2023, 00:03:55, Revision: dacb404

singularity exec /home/proteomics/mysoftware/openms/build/latest/ghcr.io-openms-openms-executables-latest.img EICExtractor -in 20231026_QC01_001_04.orig.mzML -pos 20231023_QC01_001_04.edta -out 20231023_QC01_001_04.csv

area=4.376195283497962e08
```

```
Version: 3.1.0-pre-nightly-2023-10-20 Oct 20 2023, 23:03:52, Revision: 9513f43

singularity exec /home/proteomics/mysoftware/openms/build/3.1.0/ghcr.io-openms-openms-executables-3.1.0.img EICExtractor -in 20231026_QC01_001_04.orig.mzML -pos 20231023_QC01_001_04.edta -out 20231023_QC01_001_04.3.1.0.csv

area=9.74320432e08
```

```
RT m/z int
864 582.3189 0
```

**Comment by rolivella** [comment]
Sorry I forgot the `rt_tol 240 -mz_tol 5`. If I put it, I get `1.267693787255913e09` with the last revision, which is very close to XCalibur! Good! I'll do some more tests and let you know.
